site/docs/coverage.md

```diff
@@ -0,0 +1,192 @@
+browser:
+
+```console
+genhtml --output genhtml "$(bazel info output_path)/_coverage/_coverage_report.dat"
+```
+
+Note that `genhtml` reads the source code as well, to annotate missing
+coverage in these files. For this to work, it is expected that
+`genhtml` is executed in the root of the bazel project.
+
+To view the result, simply open the `index.html` file produced in the
+`genhtml` directory in any web browser.
+
+For further help and information around the `genhtml` tool, or the
+`lcov` coverage format, see [the lcov project][lcov].
+
+## Remote execution
+
+Running with remote test execution currently has a few caveats:
+
+- The report combination action cannot yet run remotely. This is
+  because Bazel does not consider the coverage output files as part of
+  its graph (see [this issue](https://github.com/bazelbuild/bazel/issues/4685)), and can therefore not correctly treat them as inputs to
+  the combination action. To work around this, use
+  `--strategy=CoverageReport=local,remote`.
+- `--remote_download_minimal` and similar flags can also not be used
+  as a consequence of the former.
+- Bazel will currently fail to create coverage information if tests
+  have been cached previously. To work around this,
+  `--nocache_test_results` can be set specifically for coverage runs,
+  although this of course incurs a heavy cost in terms of test times.
+
+## Language-specific configuration
+
+### Java
+
+Java should work out-of-the-box with the default configuration. The
+[bazel toolchains](https://github.com/bazelbuild/bazel-toolchains)
+contain everything necessary for remote execution, as well, including
+JUnit.
+
+[lcov]: https://github.com/linux-test-project/lcov
+
+### Python
+
+#### Prerequisites
+
+Running coverage with python requires some prerequisites:
+
+- A bazel binary that includes [b01c859](https://github.com/bazelbuild/bazel/commit/b01c85962d88661ec9f6c6704c47d8ce67ca4d2a), `bazel-6.0.0` currently has this feature already.
+- A [modified version of coverage.py](https://github.com/ulfjack/coveragepy/tree/lcov-support).
+
+#### Consuming the modified coverage.py
+
+A way to do this is via [rules_python](https://github.com/bazelbuild/rules_python), this provides the ability to use a `requirements.txt` file, the requirements listed in the file are then created as bazel targets using the [pip_install](https://github.com/bazelbuild/rules_python#installing-pip-dependencies) repository rule.
+
+The `requirements.txt` should have the following entry:
+
+```text
+git+https://github.com/ulfjack/coveragepy.git@lcov-support
+```
+
+The `rules_python`, `pip_install`, and the `requirements.txt` file should then be used in the WORKSPACE file as:
+
+```python
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+http_archive(
+    name = "rules_python",
+    url = "https://github.com/bazelbuild/rules_python/releases/download/0.5.0/rules_python-0.5.0.tar.gz",
```

Comment:
Any specific version requirement?

Reply:
I believe we use this version because for compatibility with Bazel 6. This might need to be documented a bit better, although in theory this is documentation for Bazel 6.

Reply:
SG, you could add a note for that.

Reply:
Nevermind, there's no particular reason for `rules_python`, I misremembered - we use the latest `bazel_toolchain` for that reason. There doesn't seem to be a particular reason for using 0.5.0, besides it being the latest available. I'll add a note to document that.

Interestingly, changing the input url but not changing the checksum doesn't seem to break anything: https://github.com/EngFlow/kotlin-perf-analysis/actions/runs/1530749355. This seems like a major bug?